app/models/reports/schema_v2.rb

```diff
@@ -0,0 +1,230 @@
+module Reports
+  class SchemaV2
+    include BustCache
+    include Memery
+
+    attr_reader :control_rate_query_v2
+    attr_reader :periods
+    attr_reader :period_hash
+    attr_reader :period_type
+    attr_reader :regions
+
+    delegate :cache, :logger, to: Rails
+
+    def initialize(regions, periods:)
+      @regions = regions
+      @periods = periods
+      @period_type = @periods.first.type
+      @control_rate_query_v2 = ControlRateQueryV2.new
+      @period_hash = lambda { |month_date, count| [Period.month(month_date), count] }
+    end
+
+    # Returns the earliest patient record for a Region from either assigned or registered patients. Note that this *ignores*
+    # the periods that are passed in for the Repository - this is the true 'earliest report date' for a Region.
+    memoize def earliest_patient_recorded_at
+      region_entries = regions.map { |region| RegionEntry.new(region, __method__) }
+      cached_results = cache.fetch_multi(*region_entries, force: bust_cache?) { |region_entry|
+        earliest_patient_data_query_v2(region_entry.region)
+      }
+      cached_results.each_with_object({}) { |(region_entry, time), results| results[region_entry.slug] = time }
```

Comment:
Do we need to cache this?

Reply:
Hopefully not - I'll do some measurements after consolidating queries, where I think the first step is to rip out the caching from v2.